### ISSUE TITLE
nimble run doesn't preserve quoted arguments 

### ISSUE THREAD

**Issue by atstp** [nim-lang/nimble#767]
## the issue

calling 

```
$ nimble run my_binary "one argument with spaces"
```

is not the same as

```
$ nimble build my_binary
$ ./my_binary "one argument with spaces"
```

## other info 

- at first glance, [this line](https://github.com/nim-lang/nimble/blob/9391fbc56d9e4d3080c10583268f243b8b78edd5/src/nimble.nim#L1133) looks like it's a part (all?) of the issue.
- #292 doesn't have much substance, but it's the closest i could find when searching for duplicates
- my `nimble` version is `v0.11.0`
- my `nim` version is  `1.0.4`

## reproducing

I'm on debian and the following was sufficient:

```
$ nimble init count_cli_args # i tested with "Binary" for Package Type and "c" for the Package Backend
$ cd count_cli_args/
$ cat >src/count_cli_args.nim <<EOF
import os

when isMainModule:
  echo "got ", paramCount(), " command line arguments"
EOF
$ nimble run count_cli_args "One argument with spaces"
  Verifying dependencies for count_cli_args@0.1.0
   Building count_cli_args/count_cli_args using c backend
got 4 command line arguments
$ nimble build count_cli_args
  Verifying dependencies for count_cli_args@0.1.0
   Building count_cli_args/count_cli_args using c backend
$ ./count_cli_args "One argument with spaces"
got 1 command line arguments
```

**Comment by atstp** [comment]
my workaround at the moment (following on the "reproducing" section above):

```
$ mynimblerun(){ a=$1; shift; nimble build $a && ./$a "$@"; }
$ mynimblerun count_cli_args "one argument, many params"
  Verifying dependencies for count_cli_args@0.1.0
   Building count_cli_args/count_cli_args using c backend
got 1 command line arguments
```

**Comment by atstp** [comment]
If you guys can offer some general guidelines for an approach that would be likely to be merged, I'd be willing to start work on a patch.

**Comment by Araq** [comment]
Fix it please and I'll fight for you. :-)

**Comment by dom96** [comment]
If you can fix it within the `run` proc then it'll be an easy merge, if you need to modify the command args parser then it will need some more thorough review but should still be good.

**Comment by atstp** [comment]
closed with PR #771
